plugin/README.md

```diff
@@ -0,0 +1,155 @@
+	homePath string, invCheckPeriod uint, encodingConfig simappparams.EncodingConfig,
+	appOpts servertypes.AppOptions, baseAppOptions ...func(*baseapp.BaseApp),
+) *SimApp {
+
+	...
+
+	keys := sdk.NewKVStoreKeys(
+		authtypes.StoreKey, banktypes.StoreKey, stakingtypes.StoreKey,
+		minttypes.StoreKey, distrtypes.StoreKey, slashingtypes.StoreKey,
+		govtypes.StoreKey, paramstypes.StoreKey, ibchost.StoreKey, upgradetypes.StoreKey,
+		evidencetypes.StoreKey, ibctransfertypes.StoreKey, capabilitytypes.StoreKey,
+	)
+
+	pluginsOnKey := fmt.Sprintf("%s.%s", plugin.PLUGINS_TOML_KEY, plugin.PLUGINS_ON_TOML_KEY)
+	if cast.ToBool(appOpts.Get(pluginsOnKey)) {
+		// this loads the preloaded and any plugins found in `plugins.dir`
+		pluginLoader, err := loader.NewPluginLoader(appOpts, logger)
+		if err != nil {
+			// handle error
+		}
+
+		// initialize the loaded plugins
+		if err := pluginLoader.Initialize(); err != nil {
+			// handle error
+		}
+
+		// register the plugin(s) with the BaseApp
+		if err := pluginLoader.Inject(bApp, appCodec, keys); err != nil {
+			// handle error
+		}
+
+		// start the plugin services, optionally use wg to synchronize shutdown using io.Closer
+		wg := new(sync.WaitGroup)
+		if err := pluginLoader.Start(wg); err != nil {
+			// handler error
+		}
+	}
+
+	...
+
+	return app
+}
+```
+
+# State Streaming Plugin
+The `BaseApp` package contains the interface for a `StreamingService` used to write state changes out from individual KVStores to a
+file or stream, as described in [ADR-038](../docs/architecture/adr-038-state-listening.md).
+
+Specific `StreamingService` implementations are written and loaded as plugins by extending the above interface with a
+`StateStreamingPlugin` interface that adds `Register` method used to register the plugin's `StreamingService` with the
+`BaseApp` and a `Start` method to start the streaming service.
+
+```go
+// StateStreamingPlugin interface for plugins that load a streaming.Service onto a baseapp.BaseApp
+type StateStreamingPlugin interface {
+	// Register configures and registers the plugin streaming service with the BaseApp
+	Register(bApp *baseapp.BaseApp, marshaller codec.BinaryCodec, keys map[string]*types.KVStoreKey) error
+
+	// Start starts the background streaming process of the plugin streaming service
+	Start(wg *sync.WaitGroup) error
+
+	// Plugin is the base Plugin interface
+	Plugin
+}
+```
+
+A `StateStreamingPlugin` is configured from within an App using the `AppOptions` loaded from the app.toml file.
+Every `StateStreamingPlugin` will be configured within the `plugins.streaming` TOML mapping. The exact keys/parameters
+present in this mapping will be dependent on the specific `StateStreamingPlugin`, but we will introduce some standards
+here using the file `StateStreamingPlugin`:
+
+Plugin TOML configuration should be split into separate sub-tables for each kind of plugin (e.g. `plugins.streaming`).
```

Comment:
is there a case a node would want to support multiple streamers? Also would be good to see how this effects the nodes performance. I can't remember if this is blocking in anyway?